app/models/finance/billing_strategy.rb

```diff
@@ -172,7 +172,9 @@ def create_invoice_counter(period)
     #  respectively to monthly billing friendly ids (YYYY-MM-########) and to yearly billing friendly ids (YYYY-########)
     invoice_prefix = period.to_param[0..(billing_monthly? ? 6 : 3)]
 
-    InvoiceCounter.create(provider_account: account, invoice_prefix: invoice_prefix, invoice_count: 0)
+    InvoiceCounter.transaction(requires_new: true) do
+      InvoiceCounter.create(provider_account: account, invoice_prefix: invoice_prefix, invoice_count: 0)
```

Comment:
Postgres uses [SAVEPOINTs](https://www.postgresql.org/docs/current/sql-savepoint.html) instead of strictly autonomous subtransactions, and apparently it works:

```ruby
[1] pry(main)> billing_strategy = Finance::BillingStrategy.all.sample
[2] pry(main)> period = Month.parse_month('2019-01')
[3] pry(main)> InvoiceCounter.count
=> 0

[4] pry(main)> Finance::BillingStrategy.transaction do
[4] pry(main)*   billing_strategy.create_invoice_counter(period)
[4] pry(main)* end
   (0.4ms)  BEGIN
   (0.2ms)  SAVEPOINT active_record_1
  SQL (11.7ms)  INSERT INTO `invoice_counters` (`provider_account_id`, `invoice_prefix`, `created_at`, `updated_at`) VALUES (2, '2019-01', '2019-01-18 15:43:07', '2019-01-18 15:43:07')
   (0.3ms)  RELEASE SAVEPOINT active_record_1
   (3.4ms)  COMMIT
=> #<InvoiceCounter:0x00007fd33d3ef478
 id: 1,
 provider_account_id: 2,
 invoice_prefix: "2019-01",
 invoice_count: 0,
 created_at: Fri, 18 Jan 2019 15:43:07 UTC +00:00,
 updated_at: Fri, 18 Jan 2019 15:43:07 UTC +00:00>

[5] pry(main)> Finance::BillingStrategy.transaction do
[5] pry(main)*   billing_strategy.create_invoice_counter(period)
[5] pry(main)* end
   (0.4ms)  BEGIN
   (0.2ms)  SAVEPOINT active_record_1
  SQL (10.1ms)  INSERT INTO `invoice_counters` (`provider_account_id`, `invoice_prefix`, `created_at`, `updated_at`) VALUES (2, '2019-01', '2019-01-18 15:43:58', '2019-01-18 15:43:58')
   (0.6ms)  ROLLBACK TO SAVEPOINT active_record_1
  InvoiceCounter Load (0.5ms)  SELECT  `invoice_counters`.* FROM `invoice_counters` WHERE `invoice_counters`.`invoice_prefix` = '2019-01' AND `invoice_counters`.`provider_account_id` = 2 LIMIT 1
   (0.3ms)  COMMIT
=> #<InvoiceCounter:0x00007fd338566ea0
 id: 1,
 provider_account_id: 2,
 invoice_prefix: "2019-01",
 invoice_count: 0,
 created_at: Fri, 18 Jan 2019 15:43:07 UTC +00:00,
 updated_at: Fri, 18 Jan 2019 15:43:07 UTC +00:00>
```

Reply:
Nice!